Libraries/Components/SafeAreaView/SafeAreaView.js

```diff
@@ -31,24 +29,33 @@ let exported;
  * sensor housing area on iPhone X).
  */
 if (Platform.OS === 'android') {
+  let exported;
   exported = class SafeAreaView extends React.Component<Props> {
     render(): React.Node {
       const {emulateUnlessSupported, ...props} = this.props;
       return <View {...props} />;
     }
   };
+  module.exports = exported;
 } else {
   const RCTSafeAreaViewNativeComponent = require('RCTSafeAreaViewNativeComponent');
-  exported = class SafeAreaView extends React.Component<Props> {
-    render(): React.Node {
-      return (
-        <RCTSafeAreaViewNativeComponent
-          emulateUnlessSupported={true}
-          {...this.props}
-        />
-      );
-    }
+
+  const SafeAreaView = (
+    props: Props,
+    forwardedRef?: ?React.Ref<typeof RCTSafeAreaViewNativeComponent>,
+  ) => {
+    return (
+      <RCTSafeAreaViewNativeComponent
+        emulateUnlessSupported={true}
+        {...props}
+        ref={forwardedRef}
+      />
+    );
   };
-}
 
-module.exports = exported;
+  const SafeAreaViewRef = React.forwardRef(SafeAreaView);
+  /* $FlowFixMe(>=0.89.0 site=react_native_fb) This comment suppresses an error
```

Comment:
If you cast through ‘any’ then you won’t need this. Search for “: any” in the codebase to find an example.

Reply:
I think it would be 

module.exports = ((SafeAreaViewRef: any): Class<NativeComponent<Props>>)